dotnet/Simple.Sauce/SauceOptions.cs

```diff
@@ -55,83 +59,114 @@ public SauceOptions()
         public string SeleniumVersion { get; set; }
         public IList<string> Tags { get; set; }
         public string TimeZone { get; set; }
-        public string TunnelIdentifier { get; set; }
         public bool VideoUploadOnPass { get; set; }
 
-        public void WithEdge()
+        public Dictionary<string, object> ToDictionary()
         {
-            WithEdge(EdgeVersion.Latest);
+            var options = new Dictionary<string, object>();
+
+            var addAuth = AddAuthentication();
+            foreach (var option in addAuth)
+                options.Add(option.Key, option.Value);
+
+            var addOptions = AddOptions();
+            foreach (var option in addOptions)
+                options.Add(option.Key, option.Value);
+
+            return options;
         }
 
-        public void WithEdge(EdgeVersion edgeVersion)
+        private Dictionary<string, object> AddAuthentication()
         {
-            if (edgeVersion == null)
-                throw new ArgumentNullException("Please supply a valid EdgeVersion. You suplied an invalid value=>" +
-                                                edgeVersion);
-            ConfiguredEdgeOptions.BrowserVersion = edgeVersion.Value;
-            ConfiguredEdgeOptions.PlatformName = DEFAULT_PLATFORM;
+            var sauceUsername = Environment.GetEnvironmentVariable("SAUCE_USERNAME");
+            var sauceAccessKey = Environment.GetEnvironmentVariable("SAUCE_ACCESS_KEY");
+
+            return new Dictionary<string, object>
+            {
+                ["username"] = sauceUsername,
+                ["accessKey"] = sauceAccessKey
+            };
+        }
+
+        private Dictionary<string, object> AddOptions()
+        {
+            var options = JsonConvert.SerializeObject(this, JsonUtils.SerializerSettings());
+            var sauceOptions = JsonConvert.DeserializeObject<Dictionary<string, object>>(options, JsonUtils.SerializerSettings());
+
+            foreach (var timeout in Timeout?.ToDictionary())
+                sauceOptions.Add(timeout.Key, timeout.Value);
+
+            if (CustomData != null)
+                sauceOptions.Add("custom-data", CustomData);
+
+            return sauceOptions;
         }
 
-        public void WithChrome()
+        private void ConfigureOptions(string version = DEFAULT_BROWSER_VERSION, string platform = DEFAULT_PLATFORM)
         {
-            ConfiguredChromeOptions.BrowserVersion = DEFAULT_BROWSER_VERSION;
-            ConfiguredChromeOptions.PlatformName = DEFAULT_PLATFORM;
+            ConfiguredOptions.BrowserVersion = version;
+            ConfiguredOptions.PlatformName = platform;
         }
 
-        public void WithChrome(string chromeVersion)
+        public SauceOptions WithChrome(string version = DEFAULT_BROWSER_VERSION, string platform = DEFAULT_PLATFORM)
```

Comment:
Codacy found an issue: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/SauceLabs/sauce_bindings/pullRequest?prid=6319844)